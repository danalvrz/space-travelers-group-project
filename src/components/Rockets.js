import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from './Rocket';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);
  if (rockets !== undefined) {
    return (
      <div>
        {rockets.map((rocket) => (
          <Rocket key={rocket.id} rocket={rocket} />
        ))}
      </div>
    );}
  return [];
  };

export default Rockets;
