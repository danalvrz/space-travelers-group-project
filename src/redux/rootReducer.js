import { combineReducers } from 'redux';
import missions from './missions/missions';
import rockets from './rockets/rockets';

const rootReducer = combineReducers({
  missions,
  rockets,
});

export default rootReducer;
