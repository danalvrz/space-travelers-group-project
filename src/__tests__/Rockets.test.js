import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import Rockets from '../components/Rockets';

const mockStore = [];
const mockMissionsReducer = () => [{
  mission_id: 0, mission_name: 'test', mission_description: 'test', member: true,
}];
const mockRocketsReducer = () => mockStore;
const mockReducer = () => combineReducers({
  mockMissionsReducer,
  mockRocketsReducer,
});
const store = createStore(mockReducer);

describe('Rockets section', () => {
  it('renders Rockets correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Rockets />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
