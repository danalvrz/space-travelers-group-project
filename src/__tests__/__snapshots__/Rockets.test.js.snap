// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets section renders Rockets correctly 1`] = `
<div>
  <h1>
    Rockets
  </h1>
</div>
`;
